src/test/java/org/mockito/internal/runners/DefaultInternalRunnerTest.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2007 Mockito contributors
+ * This program is made available under the terms of the MIT License.
+ */
+package org.mockito.internal.runners;
+
+import org.junit.Test;
+import org.junit.runner.notification.Failure;
+import org.junit.runner.notification.RunListener;
+import org.junit.runner.notification.RunNotifier;
+import org.junit.runners.model.InitializationError;
+import org.mockito.Mock;
+import org.mockito.internal.junit.MockitoTestListener;
+import org.mockito.internal.junit.NoOpTestListener;
+import org.mockito.internal.util.Supplier;
+
+import static org.junit.Assert.*;
+
+public class DefaultInternalRunnerTest {
+
+    private final Supplier<MockitoTestListener> supplier = new Supplier<MockitoTestListener>() {
+        public MockitoTestListener get() {
+            return new NoOpTestListener();
+        }
+    };
+
+    @Test
+    public void nominal() throws InitializationError {
+        InternalRunner runner = new DefaultInternalRunner(SuccessTest.class, supplier);
+        runner.run(failIfFailure());
+    }
+
+    @Test
+    public void second_test_must_success() throws InitializationError {
+        InternalRunner failRunner = new DefaultInternalRunner(TestFailOnInitialization.class, supplier);
+        failRunner.run(new RunNotifier());
+
+        InternalRunner successRunner = new DefaultInternalRunner(SuccessTest.class, supplier);
+        successRunner.run(failIfFailure());
```

Comment:
Can we add assertions for these invocations? Would like to make sure the tests were actually executed as expected.